src/test/scala/chiselTests/Aggregate.scala

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.testers.BasicTester
+import chisel3.util.DecoupledIO
+
+class FooIO[T <: Data](n: Int, private val gen: T) extends Bundle {
+  val in = Flipped(Vec(n, new DecoupledIO(gen)))
+  val out = new DecoupledIO(gen)
+}
+
+class Foo[T <: Data](n: Int, private val gen: T) extends Module {
+  val io = IO(new FooIO(n, gen))
+
+  // incomplete function, just a placeholder
+  def foo(a: DecoupledIO[T], b: DecoupledIO[T]) = {
+    a.ready := true.B
+    b.ready := true.B
+    val out = Wire(new DecoupledIO(gen))
+
+    out.valid := true.B
+
+    val regSel = RegInit(false.B)
+    out.bits := Mux(regSel, a.bits, b.bits)
+    out.ready := a.ready
+    out
+  }
+
+  io.out <> io.in.reduceTree(foo)
+}
+
+
+class AggregateTester() extends BasicTester {
+  val dut = Module(new Foo(5, UInt(5.W)))
+  dut.io := DontCare
+  stop()
+}
+
+
+
+class AggregateSpec extends ChiselPropSpec {
+  property("Aggregate functions should preserve input/output type") {
+      assertTesterPasses { new AggregateTester() }
+  }
+}
```

Comment:
Since `.reduceTree` is a method on `Vec` (not on `Aggregate`), could you move this test to `Vec.scala` in `VecSpec`? Also it would be helpful to give more precise names to the classes, eg.

`ReduceTreeModule` instead of `Foo` and `ReduceTreeTester` instead of `AggregateTester`?

Reply:
I'll